GH Actions: various updates

jrfnl opened the pull request:
### GH Actions: fix use of deprecated set-output

GitHub has deprecated the use of `set-output` (and `set-state`) in favour of new environment files.

This commit updates workflows to use the new methodology.

Refs:
* https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
* https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files

### ~~GH Actions: update the xmllint-problem-matcher~~

~~The `xmllint-problem-matcher` action runner has released a new version which updates it to use node 16.
This gets rid of a warning which was shown in the action logs.~~

~~Note: I've [suggested to the author to use long-running branches for the action runner instead](https://github.com/korelstar/xmllint-problem-matcher/pull/7), which would make this update redundant, but no telling if or when they'll respond to that, let alone if they will follow my suggestion.~~

Refs:
* https://github.com/korelstar/xmllint-problem-matcher/releases/tag/v1.1

### 🆕 GH Actions: harden the workflow against PHPCS ruleset errors

If there is a ruleset error, the `cs2pr` action doesn't receive an `xml` report and exits with a `0` error code, even though the PHPCS run failed (though not on CS errors, but on a ruleset error).

This changes the GH Actions workflow to allow for that situation and still fail the build in that case.

---

👉🏻 Note: this won't get rid of all warning yet as a lot of predefined action runners also use `set-output`, but most of those are in the process of updating and/or have released a new version already, so the other warnings should automatically disappear over the next few weeks.

jrfnl committed:
GH Actions: fix use of deprecated `set-output`

GitHub has deprecated the use of `set-output` (and `set-state`) in favour of new environment files.

This commit updates workflows to use the new methodology.

Refs:
* https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
* https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files

jrfnl commented:
Note: PR #282 is needed to get the build to pass.

jrfnl commented:
I've removed the second commit - the action runner branch has become a long-running branch, so this update is no longer needed.

sirbrillig commented:
Great! Thank you!

sirbrillig commented:
The PHP 7.4 static analysis failure is this, for the record:

<img width="551" alt="Screen Shot 2022-10-15 at 11 44 41 AM" src="https://user-images.githubusercontent.com/2036909/195995454-a6e1cb67-a36e-458d-bc25-5eeacb27af28.png">

I don't see how your changes could have caused that so I'm guessing it's just that there was an upgrade to the analyzer. If you're fine with merging, I'm fine with that.

jrfnl commented:
> I don't see how your changes could have caused that so I'm guessing it's just that there was an upgrade to the analyzer. If you're fine with merging, I'm fine with that.

Correct. That looks to be related to an update of PHPStan, not to this PR. If you rerun CI on `2.x`, the error should also show, which would confirm that it is unrelated to this PR.

Aside from that, a fix for the PHPStan issue is ready in PR #282, so if you'd merge that first, I could rebase this PR after and you can see a clean build.

jrfnl commented:
Rebased without changes to get a passing build.

jrfnl committed:
GH Actions: harden the workflow against PHPCS ruleset errors

If there is a ruleset error, the `cs2pr` action doesn't receive an `xml` report and exits with a `0` error code, even though the PHPCS run failed (though not on CS errors, but on a ruleset error).

This changes the GH Actions workflow to allow for that situation and still fail the build in that case.

jrfnl commented:
FYI: Added a new second commit with some workflow hardening (and updated the PR description to match).

sirbrillig commented:
👍  Thank you!
